### PR TITLE
Update heat and atmosphere in mi-go base

### LINKED ISSUE #34914: Mi-go structures temperature too high for NPCs to live
# Describe the bug

Contrary to #32055 which mentions the temperature to be around 56C, the actual temperature is usually in the low 100s

The NPCs inside should be dead due to the resin cages being unable to protect them from the heat & poison gas since according to its description it lets air through

# Steps To Reproduce

1. Make a player and teleport to a mi-go building
2. Wait around 2 minutes then spawn a thermometer or divers watch
3. Activate thermometer or divers watch to see the high temperature

# Expected behavior

The temperature would be at around 56C & the NPCs would be in some kind of containment
which prevents the heat and poison gas from going through unlike the current resin cages which
probably would let it through due to being able to let air through

Ideally also describe *why* you expect it to happen

The mi-go would not want their test subjects to die to the environment before they finish their testing or whatever reason they keep them alive in there
(Or maybe they want them to die to the environment? In which case they should be long dead to the gas and heat)

# Screenshots

cddami-gotemp

# Versions and configuration

 - OS: Windows
 - Game Version: 0.D-8668-gf7c73e0 [32-bit]
 - Graphics version: Tiles
 - Mods loaded: [dda], [growable-pots], [national_guard_camp]

# Additional context

None

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Bugfixes "Adjusts the heat and toxic atmosphere of mi-go bases"```

#### Purpose of change
Fixes #34914 

It appears ZhilkinSerg finished jsonizing the heat field code only two weeks after the mi-go base went in, and I was on hiatus at the time and missed it. That's all that's been needed to fix this, so now I'm fixing it.

#### Describe the solution
Makes a tightly regulated temperature field, hot_air_sauna, that caps off in the high 60 degrees celsius but also has a high bottom end in the 50s - that's what was missing to the hot_air3 field before, it had the capacity to be much too cool. I'll have to test in summer to see how much the outside temperature affects things, but this should keep it uncomfortably hot year round without being completely fatal.

While I was in there, I also changed the concealing clouds of toxic gas in the base to the invisible, smothering atmosphere I'd intended. Instead of being poisonous it makes you tired and less combative. You're forced to choose between further overheating by wearing a gasmask, or being crippled by the toxic air.

#### Describe alternatives you've considered
I could leave it as is. Surprisingly, the blisteringly hot temperatures in the base didn't have as strong an impact on it as I'd thought: it's getting roughly the play reports I was aiming for, of being nearly impossible to brave, but not entirely. Still, the temperature is definitely not meant to be quite this high, and the fifties to sixties celsius should still be hot enough to seriously deter all but the most well prepared adventurers.

#### Testing
image
Temp is a bit low but works. The atmosphere effects don't go in stages as much as I'd like, I may tweak them later, but for now they're working well enough.

### CODE REVIEW

File: data/json/field_type.json [hunk 1]
```diff
@@ -935,6 +935,23 @@
     "phase": "gas",
     "display_field": true
   },
+  {
+    "id": "fd_hot_air_sauna",
+    "type": "field_type",
+    "legacy_enum_id": 48,
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]: You probably don't need this

```suggestion
```

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/how-to-associate-an-effect-type-with-a-field-type/22018/2

File: data/json/field_type.json [hunk 2]
```diff
@@ -935,6 +935,66 @@
...
+  },
+  {
+    "id": "fd_migo_atmosphere",
+    "type": "field_type",
+    "intensity_levels": [
+      {
+        "name": "foul-smelling air",
+        "sym": "8",
+        "dangerous": true,
+        "effect_id": "migo_atmosphere",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]: There is a slightly different syntax for fields effects after #35638 is merged:

```suggestion
        "effects": [
          {
            "effect_id": "migo_atmosphere",
            "body_part": "MOUTH",
            "intensity": 1,
            "min_duration": "5 seconds",
            "max_duration": "10 seconds",
            "immune_inside_vehicle": true
          }
```

File: data/json/field_type.json [hunk 3]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_body_part": "MOUTH",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 4]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_intensity": 1,
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 5]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_min_duration": "5 seconds",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 6]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_max_duration": "10 seconds",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 7]
```diff
@@ -935,6 +935,66 @@
...
+        "inside_immune": true
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 8]
```diff
@@ -935,6 +935,66 @@
...
+      },
+      {
+        "name": "foul-smelling air",
+        "translucency": 1,
+        "effect_id": "migo_atmosphere",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
        "effects": [
          {
            "effect_id": "migo_atmosphere",
            "body_part": "MOUTH",
            "intensity": 2,
            "min_duration": "10 seconds",
            "max_duration": "25 seconds"
          }
```

File: data/json/field_type.json [hunk 9]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_body_part": "MOUTH",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 10]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_intensity": 2,
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 11]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_min_duration": "10 seconds",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 12]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_max_duration": "25 seconds"
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 13]
```diff
@@ -935,6 +935,66 @@
...
+      },
+      {
+        "name": "foul-smelling air",
+        "effect_id": "migo_atmosphere",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
        "effects": [
          {
            "effect_id": "migo_atmosphere",
            "body_part": "MOUTH",
            "intensity": 4,
            "min_duration": "15 seconds",
            "max_duration": "25 seconds"
          }
```

File: data/json/field_type.json [hunk 14]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_body_part": "MOUTH",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 15]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_intensity": 4,
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 16]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_min_duration": "15 seconds",
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

File: data/json/field_type.json [hunk 17]
```diff
@@ -935,6 +935,66 @@
...
+        "effect_max_duration": "25 seconds"
```
Comment (ZhilkinSerg) [inline data/json/field_type.json]:
```suggestion
```

**Pull request merged**